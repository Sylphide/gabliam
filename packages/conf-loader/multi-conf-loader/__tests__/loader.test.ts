import { resolve } from 'path';
import multiConfLoader from '../src';

describe('must load the good project without profile', () => {
  test('load test', async () => {
    expect(
      await multiConfLoader({
        folder: resolve(__dirname, './fixture'),
        projectName: 'test',
      })
    ).toMatchSnapshot();
  });

  test('load test2', async () => {
    expect(
      await multiConfLoader({
        folder: resolve(__dirname, './fixture'),
        projectName: 'test2',
      })
    ).toMatchSnapshot();
  });
});

describe('must load the good project with profile', () => {
  test('load test', async () => {
    expect(
      await multiConfLoader(
        {
          folder: resolve(__dirname, './fixture'),
          projectName: 'test',
        },
        'dev'
      )
    ).toMatchSnapshot();
  });

  test('load test2', async () => {
    expect(
      await multiConfLoader(
        {
          folder: resolve(__dirname, './fixture'),
          projectName: 'test2',
        },
        'dev'
      )
    ).toMatchSnapshot();
  });
});

describe('error', () => {
  test('bad path', async () => {
    expect(
      await multiConfLoader(
        {
          folder: resolve(__dirname, './fixture/empty'),
          projectName: 'test',
        },
        'dev'
      )
    ).toMatchSnapshot();
  });

  test('with bad constant', async () => {
    await expect(
      multiConfLoader(
        {
          folder: resolve(__dirname, './fixture/badconstant'),
          projectName: 'test',
        },
        'dev'
      )
    ).rejects.toMatchSnapshot();
  });

  test('with bad application', async () => {
    await expect(
      multiConfLoader(
        {
          folder: resolve(__dirname, './fixture/badapplication'),
          projectName: 'test',
        },
        'dev'
      )
    ).rejects.toMatchSnapshot();
  });
});
