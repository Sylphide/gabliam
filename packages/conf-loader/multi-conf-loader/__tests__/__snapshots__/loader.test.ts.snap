// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error bad path 1`] = `Object {}`;

exports[`error with bad application 1`] = `
[LoaderConfigParseError: Error during parsing of file "<PROJECT_ROOT>/packages/conf-loader/multi-conf-loader/__tests__/fixture/badapplication/application.yml".
    YAMLException: bad indentation of a mapping entry at line 5, column 18:
          application:
                     ^
    ]
`;

exports[`error with bad constant 1`] = `
[LoaderConfigParseError: Error during parsing of file "<PROJECT_ROOT>/packages/conf-loader/multi-conf-loader/__tests__/fixture/badconstant/constants.yml".
    YAMLException: bad indentation of a mapping entry at line 12, column 19:
                  host: *pgHost
                      ^
    ]
`;

exports[`must load the good project with profile load test 1`] = `
Object {
  "application": Object {
    "loggerConfig": Object {
      "appenders": Object {
        "out": Object {
          "type": "stdout",
        },
      },
      "categories": Object {
        "default": Object {
          "appenders": Array [
            "out",
          ],
          "level": "debug",
        },
      },
    },
    "profile": "dev",
    "typeorm": Object {
      "connectionOptions": Array [
        Object {
          "database": "test-dev",
          "host": "localhost",
          "name": "test-dev",
          "password": "test-dev",
          "schema": "test-dev",
          "synchronize": false,
          "type": "postgres",
          "username": "test-dev",
        },
      ],
    },
  },
  "logger": Object {
    "loggerConfig": Object {
      "appenders": Object {
        "out": Object {
          "type": "stdout",
        },
      },
      "categories": Object {
        "default": Object {
          "appenders": Array [
            "out",
          ],
          "level": "debug",
        },
      },
    },
  },
  "projectName": "test",
  "testBDD": Object {
    "typeorm": Object {
      "connectionOptions": Array [
        Object {
          "database": "test-dev",
          "host": "localhost",
          "name": "test-dev",
          "password": "test-dev",
          "schema": "test-dev",
          "synchronize": false,
          "type": "postgres",
          "username": "test-dev",
        },
      ],
    },
  },
}
`;

exports[`must load the good project with profile load test2 1`] = `
Object {
  "application": Object {
    "loggerConfig": Object {
      "appenders": Object {
        "out": Object {
          "type": "stdout",
        },
      },
      "categories": Object {
        "default": Object {
          "appenders": Array [
            "out",
          ],
          "level": "debug",
        },
      },
    },
    "profile": "dev",
    "typeorm": Object {
      "connectionOptions": Array [
        Object {
          "database": "test-dev",
          "host": "localhost",
          "name": "test-dev",
          "password": "test-dev",
          "schema": "test-dev",
          "synchronize": false,
          "type": "postgres",
          "username": "test-dev",
        },
      ],
    },
  },
  "logger": Object {
    "loggerConfig": Object {
      "appenders": Object {
        "out": Object {
          "type": "stdout",
        },
      },
      "categories": Object {
        "default": Object {
          "appenders": Array [
            "out",
          ],
          "level": "debug",
        },
      },
    },
  },
  "projectName": "test2",
  "testBDD": Object {
    "typeorm": Object {
      "connectionOptions": Array [
        Object {
          "database": "test-dev",
          "host": "localhost",
          "name": "test-dev",
          "password": "test-dev",
          "schema": "test-dev",
          "synchronize": false,
          "type": "postgres",
          "username": "test-dev",
        },
      ],
    },
  },
}
`;

exports[`must load the good project without profile load test 1`] = `
Object {
  "application": Object {
    "loggerConfig": Object {
      "appenders": Object {
        "out": Object {
          "type": "stdout",
        },
      },
      "categories": Object {
        "default": Object {
          "appenders": Array [
            "out",
          ],
          "level": "debug",
        },
      },
    },
    "typeorm": Object {
      "connectionOptions": Array [
        Object {
          "database": "test-default",
          "host": "localhost",
          "name": "test-default",
          "password": "test-default",
          "schema": "test-default",
          "synchronize": false,
          "type": "postgres",
          "username": "test-default",
        },
      ],
    },
  },
  "logger": Object {
    "loggerConfig": Object {
      "appenders": Object {
        "out": Object {
          "type": "stdout",
        },
      },
      "categories": Object {
        "default": Object {
          "appenders": Array [
            "out",
          ],
          "level": "debug",
        },
      },
    },
  },
  "projectName": "test",
  "testBDD": Object {
    "typeorm": Object {
      "connectionOptions": Array [
        Object {
          "database": "test-default",
          "host": "localhost",
          "name": "test-default",
          "password": "test-default",
          "schema": "test-default",
          "synchronize": false,
          "type": "postgres",
          "username": "test-default",
        },
      ],
    },
  },
}
`;

exports[`must load the good project without profile load test2 1`] = `
Object {
  "application": Object {
    "loggerConfig": Object {
      "appenders": Object {
        "out": Object {
          "type": "stdout",
        },
      },
      "categories": Object {
        "default": Object {
          "appenders": Array [
            "out",
          ],
          "level": "debug",
        },
      },
    },
    "typeorm": Object {
      "connectionOptions": Array [
        Object {
          "database": "test-default",
          "host": "localhost",
          "name": "test-default",
          "password": "test-default",
          "schema": "test-default",
          "synchronize": false,
          "type": "postgres",
          "username": "test-default",
        },
      ],
    },
  },
  "logger": Object {
    "loggerConfig": Object {
      "appenders": Object {
        "out": Object {
          "type": "stdout",
        },
      },
      "categories": Object {
        "default": Object {
          "appenders": Array [
            "out",
          ],
          "level": "debug",
        },
      },
    },
  },
  "projectName": "test2",
  "testBDD": Object {
    "typeorm": Object {
      "connectionOptions": Array [
        Object {
          "database": "test-default",
          "host": "localhost",
          "name": "test-default",
          "password": "test-default",
          "schema": "test-default",
          "synchronize": false,
          "type": "postgres",
          "username": "test-default",
        },
      ],
    },
  },
}
`;
